base/strings/substring.jl

```diff
@@ -47,6 +47,10 @@ end
 SubString(s::AbstractString) = SubString(s, 1, lastindex(s))
 SubString{T}(s::T) where {T<:AbstractString} = SubString{T}(s, 1, lastindex(s))
 
+@propagate_inbounds view(s::AbstractString, r) = SubString(s, r)
+@propagate_inbounds maybeview(s::AbstractString, r::UnitRange{<:Integer}) = view(s, r)
```

Comment:
```suggestion
@propagate_inbounds maybeview(s::AbstractString, r::AbstractUnitRange{<:Integer}) = view(s, r)
```

Ideally this would be like this, but looks like it also needs expanding of `SubString`'s method, too:

```Julia
julia> methods(SubString)
# 6 methods for type constructor:
[1] (::Type{SubString})(s::SubString, i::Int64, j::Int64) in Base at strings/substring.jl:43
[2] (::Type{SubString})(s::AbstractString) in Base at strings/substring.jl:47
[3] (::Type{SubString})(s::T, i::Int64, j::Int64) where T<:AbstractString in Base at strings/substring.jl:38
[4] (::Type{SubString})(s::AbstractString, i::Integer) in Base at strings/substring.jl:39
[5] (::Type{SubString})(s::AbstractString, i::Integer, j::Integer) in Base at strings/substring.jl:39
[6] (::Type{SubString})(s::AbstractString, r::UnitRange{#s69} where #s69<:Integer) in Base at strings/substring.jl:40
```

May as well do that here too, yeah?